packages/tx-construction/src/types.ts

```diff
@@ -36,3 +37,14 @@ export interface InitializeTxProps {
 export interface InitializeTxPropsValidationResult {
   minimumCoinQuantities: MinimumCoinQuantityPerOutput;
 }
+
+/** Pool delegation profile inspired from CIP-17. */
```

Comment:
I'd prefer to see a precise type, as think of the refactor that will be required when it comes time to implement an exporting function, which is a critical aspect of the standard. How about this:

``` typescript
export interface Cip17Pool {
  id: Cardano.PoolId;
  name?: string;
  ticker?: Cardano.StakePoolMetadata['ticker'];
  /** Value between 0 and 1. Example: 0.3344 */
  weight: number;
}
```
 ... which is then transformed to 
 
 ``` typescript
 export interface DelegationProfilePool extends Cip17Pool {
  pool: Cardano.StakePool;
  /** Value between 0 and 1. Example: 0.3344 is 33.44% */
  percentage: Percent;
  coins: bigint;
}
```

`Cip17Pool` shouldn't be scoped to transaction construction, but `DelegationProfilePool` can be if that's the only relevant context.

Reply:
I think the problem here is that we're trying to use a single type for 2 different things:
1. cip17 is delegation preferences - provided by the application, as configured by the user.
2. PersonalWallet reports current delegation portfolio state - this is where `weights` don't work and we should use `coins` instead

Reply:
[78fb90f](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/78fb90f4eb7a044097ea5f68fb734731283a8b6d) & [63e7597](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/63e7597dea79efdd27dcf138d894f2b6801d4276): `Cip17Pool` is replaced by `StakeByPool`

Reply:
Updated types here: [d4e79ed](https://github.com/input-output-hk/cardano-js-sdk/pull/735/commits/d4e79ed6e4d7cc95b1ceaeb19ad830bfe2c058c1)